common/flatpak-installation.c

```diff
@@ -991,6 +991,55 @@ _ostree_collection_ref_free0 (OstreeCollectionRef *ref)
   ostree_collection_ref_free (ref);
 }
 
+static gboolean
+flatpak_installation_check_installed_ref_missing_related_ref (FlatpakInstallation *self,
+                                                              const gchar         *remote_name,
+                                                              gchar               *full_ref,
+                                                              GCancellable        *cancellable)
+{
+  g_autoptr(FlatpakDir) dir  = NULL;
+  g_autoptr(GPtrArray) remote_related_refs = NULL;
+  g_autoptr(FlatpakRemoteState) state = NULL;
+  g_autoptr(GError) local_error = NULL;
+  guint j;
+
+  dir = flatpak_installation_get_dir (self, &local_error);
+  if (dir == NULL)
+    return FALSE;
+
+  state = flatpak_dir_get_remote_state_optional (dir, remote_name, FALSE, cancellable, &local_error);
+  if (state == NULL)
+   {
+      g_warning ("Unable to get remote state for %s: %s", remote_name, local_error->message);
+      return FALSE;
+   }
+
+  remote_related_refs = flatpak_dir_find_remote_related (dir, state, full_ref,
+                                                         cancellable, &local_error);
+  if (remote_related_refs == NULL)
+   {
+      g_warning ("Unable to get remote related refs for %s: %s", full_ref, local_error->message);
+      return FALSE;
+   }
+
+  for (j = 0; j < remote_related_refs->len; j++)
+    {
+      FlatpakRelated *rel = g_ptr_array_index (remote_related_refs, j);
+      g_autoptr(GFile) deploy = NULL;
+
+      if (flatpak_dir_ref_is_masked (dir, rel->ref))
+          continue;
+
+      deploy = flatpak_dir_get_if_deployed (dir, rel->ref, NULL, cancellable);
+      /* If the related extension ref was meant to be auto-installed but was not found to be
+       * deployed, return TRUE. It will be pulled in via a FlatpakTransaction's update-op again. */
+      if (rel->download && deploy == NULL)
```

Comment:
Can you do `if (!rel->download) continue` before checking for the deploy, then we'd avoid unnecessary i/o for non-download extensions.

Reply:
Addressed.